conans/client/conanfile/configure.py

```diff
@@ -47,6 +47,8 @@ def run_configure_method(conanfile, down_options, down_ref, ref):
 
 def _validate_fpic(conanfile):
     v2_mode = get_env(CONAN_V2_MODE_ENVVAR, False)
+    # FIXME: The toolchain() has dissapeared, but now it is integrated with generators.
+    # FIXME: Not sure this check should be here, this could raise before the graph is fully evaluated
     toolchain = hasattr(conanfile, "toolchain")
```

Comment:
?
```suggestion
    toolchain = hasattr(conanfile, "toolchain") or hasattr(conanfile, "generate")
```

Reply:
Problem here is that ``toolchain()`` was both an attribute and a method. If it is defined, it was activating this behavior. Now, we could have the ``generate()`` method, that would fire it, but defining and using ``generators = "CMakeToolchain"`` will not. So it seemed a bit inconsistent, and I added the FIXMEs to reconsider how to approach this (it seems that this could be moved to ``validate()`` stage, when things are evaluated after the graph is computed.